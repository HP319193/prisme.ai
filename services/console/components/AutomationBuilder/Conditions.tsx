import { FC, memo, useEffect, useMemo } from 'react';
import {
  Handle,
  NodeProps,
  Position,
  useUpdateNodeInternals,
} from 'react-flow-renderer';
import Block from './Block';
import { Flow } from './flow';
import styles from './styles';

export const Conditions: FC<NodeProps> = (props) => {
  const { data, id } = props;
  const sources = useMemo(
    () => [...Object.keys(data.value || []), Flow.NEW_CONDITION],
    [data]
  );
  const updateNodeInternals = useUpdateNodeInternals();

  useEffect(() => {
    updateNodeInternals(id);
  }, [id, sources, updateNodeInternals]);

  return (
    <>
      <Handle type="target" position={Position.Top} style={styles.handle} />
      <Block {...props} type="condition" />
      <div className="flex justify-between" style={{ padding: '0 10px' }}>
        {sources.map((key) => (
          <Handle
            key={key}
            id={key}
            type="source"
            position={Position.Bottom}
            style={{
              ...styles.handle,
              display: 'flex',
              position: 'relative',
              left: 'auto',
              bottom: '5px',
            }}
          />
        ))}
      </div>
    </>
  );
};

export default memo(Conditions);
