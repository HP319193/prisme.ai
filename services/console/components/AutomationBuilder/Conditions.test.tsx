import Conditions from './Conditions';
import renderer, { act } from 'react-test-renderer';
import { useUpdateNodeInternals } from 'react-flow-renderer';

jest.mock('./context', () => {
  const mock = {
    getApp: () => ({
      name: 'App',
      icon: '/icon.svg',
    }),
  };
  return {
    useAutomationBuilder: () => mock,
  };
});

jest.mock('./Block', () => () => null);

jest.mock('react-flow-renderer', () => {
  const mock = jest.fn();
  return {
    Handle: () => <div>Handle</div>,
    Position: {
      Bottom: 0,
    },
    useUpdateNodeInternals: () => mock,
  };
});

it('should render', () => {
  const root = renderer.create(
    <Conditions
      id="a"
      data={{}}
      type="conditions"
      selected={false}
      isConnectable={false}
    />
  );
  expect(root.toJSON()).toMatchSnapshot();

  act(() => {
    root.update(
      <Conditions
        id="b"
        data={{}}
        type="conditions"
        selected={false}
        isConnectable={false}
      />
    );
  });

  expect(useUpdateNodeInternals()).toHaveBeenCalledWith('b');
});
