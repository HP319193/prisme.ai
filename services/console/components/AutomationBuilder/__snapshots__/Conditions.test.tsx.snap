// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
Array [
  <div>
    Handle
  </div>,
  <div
    className="flex justify-content-between"
    style={
      Object {
        "padding": "0 10px",
      }
    }
  >
    <div>
      Handle
    </div>
  </div>,
]
`;
